src/js/extension.js

```diff
@@ -0,0 +1,73 @@
+var Extension;
+(function(){
+
+    /* global Util */
+
+    Extension = function (options) {
+        Util.extend(this, options);
```

Comment:
For this first pass with converting things over to be extensions and use this base class, can we just make `.options` be a member variable containing the options? It'll make converting over to using this model simpler for the first pass and once we have all of the extensions made we might have different needs/requirements for how we handle options passed into extensions.